### ISSUE TITLE
Cannot copy&paste text 

### ISSUE THREAD

**Issue by fgadaleta** [zellij-org/zellij#2400]

**Comment by raphCode** [comment]
Can you please provide more information?
What did you try?
What setup: terminal emulator?  ssh?

zellij has some options regarding clipboard copy/paste, did you try them?

**Comment by fgadaleta** [comment]
CTRL + SHIFT + x to copy and CTRL + SHIFT + v to paste to console, is not working

**Comment by raphCode** [comment]
Marking text with the mouse while holding shift should produce a selection from the terminal emulator. You should be able to copy that and paste it as well with the shortcuts you mentioned.
For me, this works using alacritty.

If you want, I am happy to help you get this to work. Please provide more information / answer my questions.
Your terse writing makes it hard to understand what is going on and comes across rude.

**Comment by fgadaleta** [comment]
yes indeed it works.
Only issue is that if i have multiple panels and select a row in one panel, it selects the entire row of the screen (expected, but nice to have)

**Comment by raphCode** [comment]
clicking + dragging without any keys should activate zellij's internal selection that only spans the active pane. On mouse release zellij copies the selection to the clipboard.
You can configure in the options whether to use the primary selection or normal system clipboard.
However, this relies on clipboard support of the terminal emulator and anything between it and zellij. E.g. if you are using ssh, it must support that too.

Workaround if you don't get it to work: maximize the current pane (Ctrl-P F) and disable pane frames (Ctrl-P Z), then use shift + mouse select.

**Comment by fgadaleta** [comment]
ah this is great!
thank you ;)

**Comment by raphCode** [comment]
it works now? :)

**Comment by fgadaleta** [comment]
maximizing the panel and holding shift, it works.

(though a bit annoying to maximize/minimize the panel all the time :P )

**Comment by raphCode** [comment]
> Dragging and clicking indeed selects and copies from the same panel. But when I paste, it finds something that is in the Linux clipboard (I find something i copied from e.g. firefox)

What does it say on the bottom when you release the mouse?

For me, it's this:
![image](https://user-images.githubusercontent.com/15750438/234799993-e8c389c3-e880-4b24-b36d-c2c0f107f1c0.png)
Afterwards I can paste the text with Ctrl-V anywhere.

**Comment by fgadaleta** [comment]
Yes, it says "Text copied to system clipboard" but when i SHIFT+CTRL+V (to console) it pastes what i copied with a regular CTR + V from somewhere else. 

In console CTRL + V does not work. 
When i copy from zellij via selection and CTR + V anywhere else I find the same thing I copied out of zellij

**Comment by raphCode** [comment]
> When i copy from zellij via selection and CTR + V anywhere else I find the same thing I copied out of zellij

Very good, that means zellij and your terminal emulator are working correctly for copying!

> Yes, it says "Text copied to system clipboard" but when i SHIFT+CTRL+V (to console) it pastes what i copied with a regular CTR + C from somewhere else.

This, however, is very weird.
Since SHIFT+CTRL+V is handled by the terminal emulator, zellij is likely not at fault here. What terminal are you using?

Can you try:
- Copying in zellij as described
- switching to some other window
- (optional: paste there to make sure the clipboard really contains zellij stuff)
- switch back to the terminal window, SHIFT+CTRL+V

My reasoning is that somehow the terminal seems not to update its/the system clipboard

**Comment by fgadaleta** [comment]
exactly it seems it's not updating the system clipboard (no clue why).
when i paste here (after copying from zellij), it doesn't paste the right thing.
Same when i do shift+ctrl+v

I am using Mate Terminal
Same issue in Konsole

**Comment by raphCode** [comment]
> when i paste here (after copying from zellij), it doesn't paste the right thing.

Oh, so the previous message read like it does?
For clarification:
- click+drag in zellij for selection
- mouse release -> zellij reports "Text copied to system clipboard"
- the clipboard does still contain the old thing?

> I am using Mate Terminal
> Same issue in Konsole

They have to support the OSC52 escape sequence. I could not find if Mate supports it, but there is an open bug against Konsole that indicates that it does not support that: https://bugs.kde.org/show_bug.cgi?id=372116

You can test if your terminal emulator supports copy to system clipboard by executing:
`printf "\033]52;c;Y29weXBhc3RhIQ==\a\n"`
After that, the clipboard should contain `copypasta!`

The solution is to use a different terminal.

**Comment by fgadaleta** [comment]
![image](https://user-images.githubusercontent.com/11614447/234870964-a9812a99-a7aa-4ba4-b4eb-8f8db2a442ab.png)

but good news, with `alacritty` it works like charm. So that's definitely a terminal issue

**Comment by grm34** [comment]
I use this workaround to get paste working on CTRL+SHIFT+V
```yaml
copy_command "wl-copy -p"
```

**Comment by tonjo** [comment]
Hi, 
this issue is closed, so I'd expect copy and paste will work, but it won't,
neither with middle click nor shift-ctrl-v.

The only workaround is options --disable-mouse-mode
but I prefer not. 
Will it never be fixed?

**Comment by paolo-lulli** [comment]
Same here: in gnome-terminal paste won't work
